src/Modules/UngradedModules.jl

```diff
@@ -807,16 +908,211 @@ function base_ring(M::SubModuleOfFreeModule)
+end
+
+function std_basis(submod::SubModuleOfFreeModule, ordering::ModuleOrdering)
+  if !haskey(submod.groebner_basis, ordering)
+    gb = compute_std_basis(submod, ordering)
+    submod.groebner_basis[ordering] = gb
   end
-  return submod.std_basis
+  return submod.groebner_basis[ordering]
+end
+
+@doc Markdown.doc"""
+    groebner_basis(submod::SubModuleOfFreeModule, ordering::ModuleOrdering)
+
+Compute a Gröbner of `submod` with respect to the given `ordering`.
+The ordering must be global.
+"""
+function groebner_basis(submod::SubModuleOfFreeModule, ordering::ModuleOrdering)
+  @assert is_global(ordering)
+  return std_basis(submod, ordering)
+end
+
+@doc Markdown.doc"""
+    reduced_groebner_basis(submod::SubModuleOfFreeModule)
+
+Compute a reduced Gröbner basis. The return type is `ModuleGens`.
+"""
+function reduced_groebner_basis(submod::SubModuleOfFreeModule)
+  return reduced_groebner_basis(submod, default_ordering(submod))
+end
+
+@doc Markdown.doc"""
+    reduced_groebner_basis(submod::SubModuleOfFreeModule, ordering::ModuleOrdering)
+
+Compute a reduced Gröbner basis with respect to the given `ordering`. The return type is `ModuleGens`.
+"""
+function reduced_groebner_basis(submod::SubModuleOfFreeModule, ordering::ModuleOrdering)
+  @assert is_global(ordering)
+  
+  if !haskey(submod.groebner_basis, ordering)
+    gb = compute_std_basis(submod, ordering, true)
+    submod.groebner_basis[ordering] = gb
+    return gb
+  else
+    if !(submod.groebner_basis[ordering].is_reduced)
+      red_gb = get_attribute(submod.groebner_basis[ordering], :reduced_groebner_basis)
+      if red_gb === nothing
+        red_gb = compute_std_basis(submod, ordering, true)
+        set_attribute!(submod.groebner_basis[ordering], :reduced_groebner_basis => red_gb)
+      else
+        return red_gb
+      end
+    else
+      return submod.groebner_basis[ordering]
+    end
+  end
+end
+
+function leading_module(submod::SubModuleOfFreeModule)
+  return leading_module(submod, default_ordering(submod))
+end
+
+function leading_module(submod::SubModuleOfFreeModule, ordering::ModuleOrdering)
+  gb = std_basis(submod, ordering)
+  return SubModuleOfFreeModule(submod.F, leading_monomials(gb))
+end
+
+@doc Markdown.doc"""
+    compute_std_basis(submod::SubModuleOfFreeModule)
+
+Compute a standard basis of `submod` and return it as a `ModuleGens`.
+"""
+function compute_std_basis(submod::SubModuleOfFreeModule)
+  return compute_std_basis(submod, default_ordering(submod))
+end
+
+@doc Markdown.doc"""
+    compute_std_basis(submod::SubModuleOfFreeModule, ordering::ModuleOrdering, reduced::Bool=false)
+
+Compute a stanard basis of `submod` with respect to the given ordering.
+Allowed orderings are those that are allowed as orderings for Singular polynomial rings.
+In case `reduced` is `true` and the ordering is global, a reduced Gröbner basis is computed.
+"""
+function compute_std_basis(submod::SubModuleOfFreeModule, ordering::ModuleOrdering, reduced::Bool=false)
```

Comment:
Can't you get rid of the separate specialized method just before this one by providing a default for ordering, i.e.
```suggestion
function compute_std_basis(submod::SubModuleOfFreeModule, ordering::ModuleOrdering = default_ordering(submod), reduced::Bool=false)
```